MSProject/ViewControllers/DetailBook/WBDetailBookViewController.swift

```diff
@@ -50,33 +50,33 @@ class WBDetailBookViewController: UIViewController {
     private func initBookDetailTableViewModel() {
 
         // Rent Book
-        viewModel.rentBookAction.values.observeValues { [unowned self] _ in
-            self.showAlert(message: "BOOK_RENTED".localized())
-            self.bookViewModel.rented = true
-            self._detailHeaderView.setup(with: self.bookViewModel)
-            self.viewModel.bookAvailable.value = false
-        }
-        
-        viewModel.rentBookAction.errors.observeValues { [unowned self] error in
-            self.showAlert(message: error.localizedDescription)
-        }
-        
-        viewModel.rentBookAction.isExecuting.signal.observeValues { [unowned self] isExecuting in
-            if isExecuting {
+        _detailHeaderView.rentButton?.reactive.controlEvents(.touchUpInside)
```

Comment:
what was the problem with using an Action?

Reply:
the issue was the Action were disabled when the book were already rented and I wanted to show an alert for the user... and I have no idea how to do that so I preferred to change it for an controlEvent and a guard { }

Reply:
I guess you could have just removed the validation on the action init, that way it's always enabled, also there is no need to bind actions to buttons (it's just the most common case) you can have an action an trigger it on your code using apply()

Reply:
I'm not saying this implementation is wrong, just that you could have still used the action if you wanted.